include/jemalloc/internal/buf_writer.h

```diff
@@ -21,6 +21,13 @@ typedef struct {
 	size_t buf_size;
 	size_t buf_end;
 	bool internal_buf;
+	/*
+	 * The following fields are for advanced APIs - see explanations
```

Comment:
I think adding `when allow_auto_flush is false` is somewhat misleading. It might be slightly better to put this `advanced APIs (i.e. when allow_auto_flush is false)` below the `bool allow_auto_flush;` line, but `allow_auto_flush` is also only meant for advanced APIs. Anyway, I think `see explanations below` should suffice.